sklearn/preprocessing/tests/test_polynomial.py

```diff
@@ -552,6 +552,37 @@ def test_polynomial_features_csr_X(deg, include_bias, interaction_only, dtype):
     assert_array_almost_equal(Xt_csr.A, Xt_dense)
 
 
+@pytest.mark.parametrize("columns", [1, 2, 3, 1000])
+def test_polynomial_features_csr_wide(columns):
+    x = sparse.csr_matrix(([1], ([0], [columns])))
+
+    est = PolynomialFeatures(3)
+    try:
+        est.fit_transform(x)
+    except ValueError:
+        pytest.fail("possible overflow")
+
+
+@pytest.mark.parametrize(
+    ["n_features", "degree", "interaction_only", "include_bias"],
+    [
+        (i, j, a, b)
+        for i in range(1, 5)
+        for j in range(1, 5)
+        for a in [True, False]
+        for b in [True, False]
+    ],
+)
+def test_num_combinations(n_features, degree, interaction_only, include_bias):
```

Comment:
Would it not be sufficient to just assert that the width of the transformed matrix is equal to n_features_out_ rather than testing the private API?

Reply:
I don't have any real issue about this, although I've not checked if the size of `_combinations` is tested elsewhere in the file.